galaxy/api/tests/test_exception_handler.py

```diff
@@ -0,0 +1,67 @@
+# (c) 2012-2019, Ansible by Red Hat
+#
+# This file is part of Ansible Galaxy
+#
+# Ansible Galaxy is free software: you can redistribute it and/or modify
+# it under the terms of the Apache License as published by
+# the Apache Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# Ansible Galaxy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# Apache License for more details.
+#
+# You should have received a copy of the Apache License
+# along with Galaxy.  If not, see <http://www.apache.org/licenses/>.
+
+import operator
+
+from django.core.exceptions import PermissionDenied
+from django.http import Http404
+from rest_framework.exceptions import APIException, ValidationError
+
+from galaxy.api.base import exception_handler
+
+
+class TestExceptionHandler:
+
+    def test_http_not_found(self):
+        error = Http404('Not found.')
+        response = exception_handler(error, None)
+        assert response.data == {
+            'message': 'Not found.',
+        }
+
+    def test_http_permission_denied(self):
+        error = PermissionDenied('Permission denied.')
+        response = exception_handler(error, None)
+        assert response.data == {
+            'message': 'Permission denied.',
+        }
+
+    def test_api_exc_string(self):
```

Comment:
Maybe test cases for other detail types?  `ApiException(detail=None)`, `ApiException(detail={'some_key':'some_value'})`, `ApiException(detail=rest_framework.utils.serializer_helpers.ReturnList(serializer=SomeSerializer))`?

Reply:
`ReturnList` and `ReturnDict` classes are subclasses of `list` and `dict` respectively and add additional reference to `serializer`, which is not uses during error serialization.